R/Ameriflux_data_tables.R

```diff
@@ -0,0 +1,562 @@
+                  skip = 2,
+                  stringsAsFactors = F)
+
+data3 <- read.csv('/home/reevesman/Ameriflux/AMF_US-ICs/AMF_US-ICs_BASE_HH_2-1.csv',
+                  skip = 2,
+                  stringsAsFactors = F)
+
+data4 <- read.csv('/home/reevesman/Ameriflux/AMF_US-ICh/AMF_US-ICh_BASE_HH_2-1.csv',
+                  skip = 2,
+                  stringsAsFactors = F)
+
+data5 <- read.csv('/home/reevesman/Ameriflux/AMF_US-EML/AMF_US-EML_BASE_HH_1-4.csv',
+                  skip = 2,
+                  stringsAsFactors = F)
+
+data6 <- read.csv('/home/reevesman/Ameriflux/AMF_US-Brw/AMF_US-Brw_BASE_HH_2-1.csv',
+                  skip = 2,
+                  stringsAsFactors = F)
+
+data7 <- read.csv('/home/reevesman/Ameriflux/AMF_US-Atq/AMF_US-Atq_BASE_HH_1-1.csv',
+                  skip = 2,
+                  stringsAsFactors = F)
+
+data8 <- read.csv('/home/reevesman/Ameriflux/AMF_CA-Qfo/AMF_CA-Qfo_BASE_HH_1-1.csv',
+                  skip = 2,
+                  stringsAsFactors = F)
+
+data9 <- read.csv('/home/reevesman/Ameriflux/AMF_CA-Ojp/AMF_CA-Ojp_BASE_HH_1-1.csv',
+                  skip = 2,
+                  stringsAsFactors = F)
+
+data10 <- read.csv('/home/reevesman/Ameriflux/AMF_CA-Oas/AMF_CA-Oas_BASE_HH_1-1.csv',
+                   skip = 2,
+                   stringsAsFactors = F)
+
+data11 <- read.csv('/home/reevesman/Ameriflux/AMF_CA-Man/AMF_CA-Man_BASE_HH_2-1.csv',
+                   skip = 2,
+                   stringsAsFactors = F)
+
+data12 <- read.csv('/home/reevesman/Ameriflux/AMF_CA-Gro/AMF_CA-Gro_BASE_HH_1-1.csv',
+                   skip = 2,
+                   stringsAsFactors = F)
+
+
+##############################################################################
+
+#function definition for attribute_definitions
+  #inputs:
+    #data: a dataframe with attributes to define
+    #definitions: dataframe in the style of definitions.csv
+  #output: a vector of attribute definitions to be used in attribute table
+
+##############################################################################
+
+attribute_definitions <- function(data, definitions){
+  
+  attributes <- colnames(data)
+  
+  #will be vector of attribute definitions
+  defs <- vector(mode = 'character', length = length(attributes))
+  
+  #i will be index at which to populate defs
+  i <- 1
+  #for each attribute
+  for (att in attributes){
+    
+    #if the attribute has one of the attached qualifiers
+    #get definition of qualifier to paste onto regular definition later
+    #set att to the regular attribute name (w/out qualifier)
+    #set a flag that this attribute includes a qualifier
+    
+    #did not do all of the possible combinations of qualifiers
+    #because not all possible combinations appeared in data
+    
+    if (str_sub(att,-5,-1) == '_PI_F'){
+      x <- str_sub(att,-5,-1)
+      extra <- paste(definitions[which(definitions$uniqueAttributeLabel == '_PI'), 'uniqueAttributeDefinition'],
```

Comment:
this is something i mentioned to sharis as well.  At the beginning of your function you could have a line that sets new column names for your attributes.  These column names in the attributes csv are pretty bad and make the R card harder to read.   
```
colnames(attributes) <- c("category", "name", "defintion", "units", "SI_units")
```